django/db/models/fields/related.py

```diff
@@ -294,6 +294,13 @@ def resolve_related_class(model, related, field):
                 field.do_related_class(related, model)
             lazy_related_operation(resolve_related_class, cls, self.remote_field.model, field=self)
 
+    def get_filter_kwargs_for_foreign_object(self, obj):
+        """
+        Return a dict that when passed as kwargs to self.model.filter(), yields
+        all instances that are related through this field to the foreign obj.
+        """
+        return {self.name: obj.pk}
```

Comment:
This is probably a wrong default implementation - it doesn't take into account that the relation can be to a different model. Also, we do support multicolumn relations internally, so something that uses all relation fields would be useful.

I guess this duplicates work already done in descriptors. The __get__ implementations need to build filters to get the related object(s) for an instance, so we need to check that we don't have duplicate implementations.